### ISSUE TITLE
NOISY DATASET AND OVERFITTING?

### ISSUE THREAD

**Issue by TungyuYoung** [francesclluis/direction-ambisonics-source-separation#1]
Dear Lluís,

NICE JOB! I ran into some problems when I tried to repeat your work.

1. I find that the dataset generated via "mix.py train 10000 3000 5 /home/username/Project/musdb18 --dataset musdb --render_room --room_size_range 2 2 1 --rt_range 0.2" contains some of NOISY data which sound like high frequency guassion noise signal.
2. I ran train.py with the settings you provided except for batch_size because my GPU memory is not enough. And I noticed that the model never seemed to fit and the train loss values have been fluctuating in the range of 0.02 to 0.04. What's more, 
3. To increase the speed of the experiment, I used a smaller dataset containing 1500 training sets and 150 test sets. The other settings are consistent with those mentioned above. I found that by the beginning of the 30th epoch, most of the output is close to blank audio.

I sincerely ask for your advice as to the reasons for these problems. Why do high frequency noise signals appear? Why does the training seem to seem less than optimal? Why is the loss function used an L1 loss function instead of just using SI-SDR or SSR?

I would appreciate it if you could answer my questions. Looking forward to hear from you soon.

**Comment by francesclluis** [comment]
Dear @TungyuYoung 

Apologies for my late reply.

Regarding your first question, I generated data using your command, and the generated data appeared to be fine. Could you please share some of the noisy audios you generated?

As for the training issues, I haven't encountered those problems before. Suboptimal performance might be due to the data generation process. If the initial data quality is not fine, the model will struggle to learn meaningful patterns. Additionally, I would recommend sticking to the original batch size and, if GPU memory is insufficient, consider using gradient accumulation.

Concerning the loss function, we utilized L1 as it was originally proposed for Demucs v1, as detailed in this paper:[ https://arxiv.org/abs/1911.13254](https://arxiv.org/abs/1911.13254).

**Comment by TungyuYoung** [comment]
Dear @francesclluis 

Thanks for your reply.

Regarding the phenomenon of high-frequency Guassion noise signal, I think it is the random reading of some audio clips, which happen to be "blank" clips (in fact, there are still extremely weak energy signals), when convolved with the RIR signal, it will be amplified into the so-called high-frequency Gaussian noise signal. Therefore, I made some modifications, such as determining whether the energy of an audio clip reaches a certain threshold before it is used to synthesize ambisonics.

About overfitting, that is, most of the output is close to blank audio. I didn't find out the true reason. Thus, I set the [num_mixes_with_silent_sources] to 0. And I successfully reproduced similar results.

Best.

**Comment by TungyuYoung** [comment]
@francesclluis 
Dear Francesclluis,

I sincerely ask that the how to draw the same diagram as Figure 4 in your article? Can you give me a copy of the code to draw the diagram? Can you kindly offered me the code?

Best.
